lib/kvdb/indent.c

```diff
@@ -0,0 +1,213 @@
+int long_variable_integer;
+void *long_variable_pointer;
+struct foobar_foobar;
+void * const *yoyo;
+unsigned int uintvec[32];
+
+struct foo {
+    char foobar_foobar_foobar[1234];
+    int *foo;
+    unsigned long **bar;
+};
+
+
+// Func Decls
+int
+function1(unsigned int *arg1);
+
+int
+function2(unsigned int *arg1, unsigned int **arg2, unsigned int ***arg3, unsigned int arg4);
+
+int
+function3(
+    unsigned int *arg1,
+    unsigned int **arg2,
+    unsigned int ***arg3,
+    unsigned int arg4,
+    unsigned int arg5)
+{
+    *arg1 = 0;
+    **arg2 = 1;
+    return 0;
+}
+
+
+// Func Decls w/ Attributes
+
+void HSE_NORETURN
+fn_attr1(void);
+
+HSE_NORETURN void
+fn_attr2(void);
+
+void HSE_PRINTF(1, 2)
+fn_attr3(char *fmt);
+
+void HSE_NORETURN
+fn_attr4(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar *foobar);
+
+HSE_NORETURN void
+fn_attr5(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar *foobar);
+
+void HSE_PRINTF(1, 2)
+fn_attr6(unsigned int arg1, unsigned int arg2, unsigned int arg3, struct foobar_foobar *foobar);
+
+
+hse_err_t
+long_if_expr(int flags)
+{
+    // simple long expr
+    if (long_variable_boolean || long_variable_integer > 0 || long_variable_pointer != NULL ||
+        (flags & HSE_KVS_PUT_VCOMP_MASK) == HSE_KVS_PUT_VCOMP_MASK)
+        return merr(EINVAL);
+
+    // long expr w/ HSE_LIKELY, which looks like a function call to clang-format
+    if (HSE_UNLIKELY(
+            long_variable_boolean || long_variable_integer > 0 || long_variable_pointer != NULL ||
+            (flags & HSE_KVS_PUT_VCOMP_MASK) == HSE_KVS_PUT_VCOMP_MASK))
+        return merr(EINVAL);
+
+    // long expr w/ function call in the middle
+    if (long_variable_boolean ||
+        some_long_function_name(
+            long_variable_boolean || long_variable_integer > 0 || long_variable_pointer != NULL))
+        return merr(EINVAL);
+
+    return 0;
+}
+
+
+void
+long_fn_call(void)
+{
+    // short
+    longish_printf("%u %u %u\n", 1, 2, 3);
+
+    // medium
+    longish_printf(
+        "test123=%d test123=%d test123=%d\n", 1000 + 1000 + 1000, 1000 + 1000 + 1000, 1000 + 1000);
+
+    // long
+    longish_printf(
+        "test123=%d test123=%d test123=%d\n", long_variable_integer, long_variable_integer,
+        long_variable_integer);
+
+    // without eol comment
+    printf(
+        "kblk_alloc_ms=%u kblk_write_ops=%u kblk_write_sz=%u "
+        "kblk_write_ms=%u vblk_alloc_ops=%u vblk_alloc_sz=%u "
+        "vblk_alloc_ms=%u vblk_write_ops=%u vblk_write_sz=%u\n",
+        uintvec[0], uintvec[1], uintvec[2], uintvec[3], uintvec[4], uintvec[5], uintvec[6],
+        uintvec[7], uintvec[8]);
+
+    // with eol hack
+    printf(
+        "kblk_alloc_ms=%u kblk_write_ops=%u kblk_write_sz=%u "
+        "kblk_write_ms=%u vblk_alloc_ops=%u vblk_alloc_sz=%u "
+        "vblk_alloc_ms=%u vblk_write_ops=%u vblk_write_sz=%u\n",
+        uintvec[0], uintvec[1], uintvec[2], // eol
```

Comment:
First I learned of the eol comment. Good to know.

Reply:
Yeah, end of line comments are pretty much the only way to prevent clang-format's greedy behavior, but it's ugly and noisy.  Sucks having the tool dictate how we write code, should be the other way around...

Reply:
Nothing magic about `eol`.  Any comment will do.